Sources/Markdown/Walker/Walkers/MarkupFormatter.swift

```diff
@@ -492,15 +492,23 @@ public struct MarkupFormatter: MarkupWalker {
 
     /// Get the numeral prefix for a list item if its parent is an ordered list.
     func numeralPrefix(for listItem: ListItem) -> String? {
-        guard listItem.parent is OrderedList else {
+        guard let list = listItem.parent as? OrderedList else {
             return nil
         }
         let numeral: UInt
         switch formattingOptions.orderedListNumerals {
         case let .allSame(n):
-            numeral = n
+            if list.start == 1 {
+                numeral = n
+            } else {
+                numeral = UInt(list.start)
```

Comment:
Is this right? Should we not preserve the numbering provided to `allSame`?

Reply:
Looking back at this, the existing implementation should defer to the indices given to `allSame` and `incrementing`. However, now that ordered lists can store their own starting index, we should allow this API to defer to that. I filed https://github.com/apple/swift-markdown/issues/76 to track that. That can be taken care of in a separate PR, or handled here.